admin/code/LeftAndMain.php

```diff
@@ -1442,6 +1442,8 @@ public function getSilverStripeNavigator() {
 	public function currentPageID() {
 		if($this->request->requestVar('ID') && is_numeric($this->request->requestVar('ID')))	{
 			return $this->request->requestVar('ID');
+		} elseif ($this->request->requestVar('CMSMainCurrentPageID') && is_numeric($this->request->requestVar('CMSMainCurrentPageID'))) {
```

Comment:
I think this line needs a comment to help future maintainers know WHY it's there. Maybe a `// {@see GridFieldDetailForm} for use case.`